include/aws_iot_mqtt_client_interface.h

```diff
@@ -58,154 +58,242 @@ extern "C" {
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_resubscribe] */
 IoT_Error_t aws_iot_mqtt_resubscribe(AWS_IoT_Client *pClient);
+/* @[declare_mqtt_resubscribe] */
 
 /**
- * @brief Unsubscribe to an MQTT topic.
+ * @brief Unsubscribe from an MQTT topic filter.
  *
- * Called to send an unsubscribe message to the broker requesting removal of a subscription
- * to an MQTT topic.
- * @note Call is blocking.  The call returns after the receipt of the UNSUBACK control packet.
+ * This function removes an MQTT subscription previously set by @ref mqtt_function_subscribe.
+ * It sends an MQTT UNSUBSCRIBE packet to the server and removes the topic's message
+ * handler stored by the client.
  *
- * @param pClient Reference to the IoT Client
- * @param pTopicFilter Topic filter of subscription
- * @param topicFilterLen Length of the topic filter
+ * @param[in] pClient MQTT client context
+ * @param[in] pTopicFilter Topic filter of the subscription to remove
+ * @param[in] topicFilterLen Length of topic filter to remove
  *
- * @return An IoT Error Type defining successful/failed unsubscribe call
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_unsubscribe] */
 IoT_Error_t aws_iot_mqtt_unsubscribe(AWS_IoT_Client *pClient, const char *pTopicFilter, uint16_t topicFilterLen);
+/* @[declare_mqtt_unsubscribe] */
 
 /**
- * @brief Disconnect an MQTT Connection
+ * @brief Disconnect an MQTT session.
  *
- * Called to send a disconnect message to the broker.
+ * This function sends the MQTT DISCONNECT packet, which closes the MQTT session
+ * between the client and server. After this function returns, the MQTT client
+ * context should be either freed with @ref mqtt_function_free or reopened with
+ * @ref mqtt_function_connect.
  *
- * @param pClient Reference to the IoT Client
+ * @param[in] pClient MQTT client context
  *
- * @return An IoT Error Type defining successful/failed send of the disconnect control packet.
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_disconnect] */
 IoT_Error_t aws_iot_mqtt_disconnect(AWS_IoT_Client *pClient);
+/* @[declare_mqtt_disconnect] */
 
 /**
- * @brief Yield to the MQTT client
+ * @brief Provide time for the MQTT client to process events.
  *
- * Called to yield the current thread to the underlying MQTT client.  This time is used by
- * the MQTT client to manage PING requests to monitor the health of the TCP connection as
- * well as periodically check the socket receive buffer for subscribe messages.  Yield()
- * must be called at a rate faster than the keepalive interval.  It must also be called
- * at a rate faster than the incoming message rate as this is the only way the client receives
- * processing time to manage incoming messages.
+ * This function processes the following events:
+ * - Incoming messages from the server <br>
+ * The server will send messages to clients whose subscriptions match the message's
```

Comment:
Suggested text:

Whenever a client publishes a message on a topic, the server sends that
message to all the clients whose subscriptions match the message's
topic. The messages sent by the server are received by this function,
which in turn calls the corresponding message handler. This function
must be called at a rate faster than the incoming messages, as it is the
only way the client receives processing time to manage incoming messages.

Reply:
Changed